lib/restaurant.rb

```diff
@@ -0,0 +1,34 @@
+require 'forwardable'
+
+class Restaurant
+
+  extend Forwardable
+
+  def_delegator :@menu, :list_items, :menu
+  def_delegator :@order_klass, :new, :new_order
+
+  def initialize menu_klass=Menu, order_klass=Order
+    @menu_klass = menu_klass
+    @order_klass = order_klass
+    @menu = @menu_klass.new
+    @orders = []
+  end
+
+  def change_menu new_menu
+    @menu = @menu_klass.new new_menu
+  end
+
+  def place_order customer, order_details
+    @orders << new_order(customer, menu, order_details)
+    if customer.bill_correct? self
```

Comment:
Style/GuardClause: Use a guard clause instead of wrapping the code inside a conditional expression. (https://github.com/bbatsov/ruby-style-guide#no-nested-conditionals)